tests/Framework.py

```diff
@@ -418,14 +419,26 @@ def ignoreWarning(self, category=Warning, module=""):
             warnings.filterwarnings("ignore", category=category, module=module)
             yield
 
+    @contextlib.contextmanager
+    def replayData(self, filename: str):
+        previous = self.__customFilename
+        self.__customFilename = filename
+        try:
+            yield
+        finally:
+            self.__customFilename = previous
+
     def __openFile(self, mode):
-        for _, _, functionName, _ in traceback.extract_stack():
-            if functionName.startswith("test") or functionName == "setUp" or functionName == "tearDown":
-                if functionName != "test":  # because in class Hook(Framework.TestCase), method testTest calls Hook.test
-                    fileName = os.path.join(
-                        self.replayDataFolder,
-                        f"{self.__class__.__name__}.{functionName}.txt",
-                    )
+        fileName = None
+        if self.__customFilename:
+            fileName = self.__customFilename
+        else:
+            for _, _, functionName, _ in traceback.extract_stack():
+                if functionName.startswith("test") or functionName == "setUp" or functionName == "tearDown":
+                    # because in class Hook(Framework.TestCase), method testTest calls Hook.test
+                    if functionName != "test":
+                        fileName = f"{self.__class__.__name__}.{functionName}.txt"
+        fileName = os.path.join(self.replayDataFolder, fileName) if fileName else None
```

Comment:
>_possible bug_

**✨ AI Suggestion:** The code attempts to join `self.replayDataFolder` with `fileName` even when `fileName` might be `None`. If the loop in lines 436-440 does not assign a value to `fileName`, it remains `None` (from line 432). Passing `None` to `os.path.join` will raise a `TypeError`. The conditional assignment `if fileName else None` does not prevent this, as `os.path.join` is called regardless within the conditional expression's 'true' branch.
```suggestion
                                        fileName = f"{self.__class__.__name__}.{functionName}.txt"
                        fileName = os.path.join(self.replayDataFolder, fileName) if fileName is not None else None
```